src/android/app/src/main/java/org/citron/citron_emu/utils/AmiiboKeyManager.kt

```
@@ -0,0 +1,104 @@
+            if (!keysDirectory.exists() && !keysDirectory.mkdirs()) {
+                return@withContext Result.UnableToWrite
+            }
+
+            val temporary = try {
+                File.createTempFile("key_retail_", ".tmp", keysDirectory).apply {
+                    writeBytes(keyData)
+                }
+            } catch (e: IOException) {
+                Log.error("[AmiiboKeyManager] Failed to stage key file: ${e.message}")
+                return@withContext Result.UnableToWrite
+            } catch (e: SecurityException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)